src/service.rs

```diff
@@ -61,7 +62,7 @@ pub struct Service<T> {
 
     multi_transport: MultiTransport,
 
-    listens: Vec<(Multiaddr, MultiIncoming)>,
```

Comment:
I think the simple address is very small and should not require arc, just clone is ok.

Moreover, the usage rate of this address is very low, I think there should be no problem with a direct clone.